concordium-consensus/tests/scheduler/SchedulerTests/Helpers.hs

```diff
@@ -1,24 +1,430 @@
+import Concordium.GlobalState.Types
+import Concordium.Logger
+import Concordium.Scheduler
+import qualified Concordium.Scheduler.DummyData as DummyData
+import qualified Concordium.Scheduler.EnvironmentImplementation as EI
+import qualified Concordium.Scheduler.Runner as SchedTest
+import Concordium.Scheduler.TreeStateEnvironment
 import qualified Concordium.Scheduler.Types as Types
-import Data.Word
+import Concordium.TimeMonad
 
-getResults :: [(a, TransactionSummary)] -> [(a, ValidResult)]
-getResults = map (\(x, r) -> (x, tsResult r))
+getResults :: [(a, Types.TransactionSummary)] -> [(a, Types.ValidResult)]
+getResults = map (\(x, r) -> (x, Types.tsResult r))
 
 -- | The cost for processing a simple transfer (account to account)
 -- with one signature in the transaction.
 --
 -- * @SPEC: <$DOCS/Transactions#transaction-cost-header-simple-transfer>
-simpleTransferCost :: Energy
+simpleTransferCost :: Types.Energy
 simpleTransferCost = Cost.baseCost (Types.transactionHeaderSize + 41) 1 + Cost.simpleTransferCost
 
-simpleTransferCostWithMemo1 :: Word64 -> Energy
-simpleTransferCostWithMemo1 memoSize = Cost.baseCost (Types.transactionHeaderSize + 41 + 2 + memoSize) 1
+simpleTransferCostWithMemo1 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo1 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+
+simpleTransferCostWithMemo2 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo2 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+        + Cost.simpleTransferCost
+
+-- | Monad that implements the necessary constraints to be used for running the scheduler.
+newtype PersistentBSM pv a = PersistentBSM
+    { _runPersistentBSM ::
+        BS.PersistentBlockStateMonad
+            pv
+            (BS.PersistentBlockStateContext pv)
+            (Blob.BlobStoreM' (BS.PersistentBlockStateContext pv))
+            a
+    }
+    deriving
+        ( Applicative,
+          Functor,
+          Monad,
+          BS.AccountOperations,
+          BS.ContractStateOperations,
+          BS.ModuleQuery,
+          MonadProtocolVersion,
+          BlockStateTypes,
+          Blob.MonadBlobStore,
+          BS.BlockStateQuery,
+          BS.BlockStateOperations,
+          MonadIO,
+          MonadCache BS.ModuleCache,
+          BS.BlockStateStorage
+        )
+
+deriving instance
+    (Types.AccountVersionFor pv ~ av) =>
+    MonadCache (BS.AccountCache av) (PersistentBSM pv)
+
+instance MonadLogger (PersistentBSM pv) where
+    logEvent _ _ _ = return ()
+
+instance TimeMonad (PersistentBSM pv) where
+    currentTime = return $ read "1970-01-01 13:27:13.257285424 UTC"
+
+-- |Call a function for each protocol version, returning a list of results.
+-- This is used to run a test against every protocol version.
+forEveryProtocolVersion ::
+    (forall pv. (Types.IsProtocolVersion pv) => Types.SProtocolVersion pv -> String -> a) ->
+    [a]
```

Comment:
Perhaps point out that the return value from the test must be independent of the protocol version (this is clear from the type, but I bet that it will be an issue for people less familiar with the style), and so the result of the test must be a "simple" value.